src/components/MainPage.jsx

```diff
@@ -1,29 +1,30 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect } from 'react';
 import { Header } from 'semantic-ui-react';
 import Articles from '../modules/Articles';
 import ArticleCard from './ArticleCard';
 import BreakingNews from './layout/BreakingNews';
+import { useSelector } from 'react-redux';
 
 const MainPage = () => {
-  const [articles, setArticles] = useState([]);
-  const [errorMessage, setErrorMessage] = useState('');
+  const { error, message, articles } = useSelector((state) => state);
 
   useEffect(() => {
-    Articles.index(setArticles, setErrorMessage)
+    Articles.index();
   }, []);
 
   let articleList = articles.slice(1).map((article, index) => {
-    return <ArticleCard article={article} index={index} />;
+    return <ArticleCard article={article} index={index} key={index} />;
   });
 
   return (
     <>
-      {errorMessage && 
+      {error && (
         <Header data-cy='error-message' color='red'>
-          {errorMessage}
+          {message}
```

Comment:
why is it message ut not errorMessage?

Reply:
With the Redux changes, the message appears when the error is true. I understand the confusion and we can look can probably change this going forward. With how the error messages are set up they will be used application wide.

Reply:
Thanks for the explanation!